### PR TITLE
SHA display touchups

### PR DESCRIPTION
**Pull request by rusackas**
### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
The SHA displayed in the user menu was overhwelmingly long. This now shows only the last 8 characters for reference (1 more than github!). It also adds a colon after "Version" for consistency, and a little more line height and bottom padding on the fine print so it's less claustrophobic.

Open TODO: make the fine print act like a little menu item, and copy the full SHA on click.
Open TODO?: move the "admin user" tooltip seen in the screenshot into the menu. Not sure if we need the tooltip. Thoughts, anyone?

image

### CODE REVIEW

File: superset/config.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def _try_json_readversion(filepath):
 def _try_json_readsha(filepath):
     try:
         with open(filepath, "r") as f:
-            return json.load(f).get("GIT_SHA")
+            return json.load(f).get("GIT_SHA")[-8:]
```
Comment (dpgaspar) [inline superset/config.py]: I would prefer the first 8 chars `[:8]`

**Review by dpgaspar — APPROVED** [review]
Small comment but LGTM

File: superset/config.py (hunk 1 above)
Comment (rusackas) [inline superset/config.py]:
```suggestion
            return json.load(f).get("GIT_SHA")[:8]
```

Comment (rusackas) [reply superset/config.py]: Done :)

**Comment by mistercrunch** [comment]
Is there a security issue here? Assuming we'll have a list of CVE out, it'd be pretty easy for someone to get a list of active vulnerabilities, not that it'd be much harder to test for all CVE.

I know for things like Apache (web server) or nginx, best practice is to keep that obscure to make it harder for black hats to scan the web looking for particular vulnerabilities. 

Two ways to potentially mitigate: only allow for admin, or put behind a feature flag.

@craig-rueda what dyou think?

**Comment by craig-rueda** [comment]
Depends on context - I'd prefer if this were configurable - i.e. have a config flag that allows end users to toggle the display of SHAs at all. I see that this change simply shortens said string, so toggling it is out of scope for this PR.

**Comment by mistercrunch** [comment]
Oh right, about SHA length, 8 is a little low:
https://stackoverflow.com/questions/18134627/how-much-of-a-git-sha-is-generally-considered-necessary-to-uniquely-identify-a

**Comment by rusackas** [comment]
> Oh right, about SHA length, 8 is a little low:
> https://stackoverflow.com/questions/18134627/how-much-of-a-git-sha-is-generally-considered-necessary-to-uniquely-identify-a

Haha... it was exactly that article that led me to believe 8 was enough! I just upped it to 10, but let me know if you think more would be useful.

**Review by mistercrunch — APPROVED** [review]

**Pull request merged**